### DISCUSSION TITLE
Uninstall script Linux

### DISCUSSION THREAD

**Discussion by Verhoeckx** [zed-industries/zed#14267] [Feature Requests]
Where can I find the uninstall script to uninstall Zed from Linux (Fedora)?
If you provide an install script, you should also provide an uninstall script!

**Comment by Verhoeckx** [comment]
@vrnko: thanks!
I have done the same thing: looked in the shell script and removed all files and directories that were created during the install.

Personally, I find that if you (with you I mean in this case Zed Industries) provide an install script, you should also provide an uninstall script! You can't expect some to keep you application indefinitely.

**Comment by oovaa** [comment]
i also did this 

 omar@Latitude-5480  ~  find . -name 'zed'                                                    ✔  292  11:49:40 
./.cache/zed
find: ‘./.bun/.bun-crash’: Permission denied
./.config/zed
./.local/bin/zed
./.local/share/zed
 omar@Latitude-5480  ~  cat ./.config/zed                                                   1 ↵  293  11:49:54 
cat: ./.config/zed: Is a directory
 omar@Latitude-5480  ~  ls ./.config/zed                                                    1 ↵  294  11:50:04 
settings.json  themes
 omar@Latitude-5480  ~  rm -rf ./.config/zed ./.cache/zed ./.local/bin/zed ./.local/share/zed

**Comment by rmoyano** [comment]
I ended up removing the following directories, after analyzing _https://zed.dev/install.sh_ script:

` user@notebook:~$ rm -r .local/zed.app/ .local/bin/zed .local/share/zed/ .local/share/applications/dev.zed.Zed.desktop .config/zed/ .cache/zed/`

**Reply by oovaa** [reply]
did the same

**Reply by tokisuno** [reply]
just did the same. super weird how there's no guide to uninstall for zed anywhere. why provide an install script without an uninstall one? super weird imo

**Reply by Verhoeckx** [reply]
I absolutely agree!

**Comment by liel-almog** [comment]
If a software does not have an uninstall guide you know something is up...

**Reply by tsanyqudsi** [reply]
Yes, it means that you should help build one.

**Comment by rogue-kitten** [comment]
in case someone is wondering how to do this on ubuntu: 


- It installed a symbolic link to the executable in ~/.local/bin: remove it with:
`rm $HOME/.local/bin/zed # remove symlink to the executable`

- It installed a .desktop launcher in $HOME/.local/share/applications/. The name is dev.zed.Zed.desktop, or at least starts with dev.zed.Zed. Delete that file.

- It created a directory "$HOME/.local/zed.app (or zed*.app depending on the version) and unpacked installation files in there. Delete that directory and its contents.

- Finally, you added the line export PATH=$HOME/.local/bin:$PATH to your ~/.bashrc (or .zshrc) file. Open that file with a text editor and delete that line.

**Reply by abhigyantrips** [reply]
this answer comes from an askubuntu thread, available here: https://askubuntu.com/questions/1520457/how-do-i-uninstall-zed-editor-from-my-ubuntu-24

**Comment by amithm7** [comment]
Have opened a PR to include this functionality in the install script

gh-16660

It should be easy as

```sh
curl https://zed.dev/install.sh | sh -s uninstall
```

once merged.

---

If anyone wants to use it now, this should work:

```sh
curl https://raw.githubusercontent.com/zed-industries/zed/82ad7f2b/script/install.sh | sh -s uninstall
```

**Reply by nnikolov3** [reply]
![image](https://github.com/user-attachments/assets/00417a94-a702-4ec6-a579-ecc14f0c66e5)

I believe it worked well on Fedora 40

**Reply by leematthewjackson** [reply]
Worked for me too!

**Reply by GR1MZ0** [reply]
this worked for me thanks

**Reply by mandarvaze** [reply]
Worked on Ubuntu 24.04 (Tuxedo OS 4 in reality)
I  wish this was merged till official/good solution is available. :man_shrugging: 

Thanks @amithm7

**Reply by The-Maize** [reply]
Yep works for me too, that was simple. well done.
Tuxedo OS 4

**Reply by mandarvaze** [reply]
Hey @The-Maize Good to know another Tuxedo OS 4 user :smile:

**Reply by The-Maize** [reply]
> Hey @The-Maize Good to know another Tuxedo OS 4 user 😄

We are a rare breed lol. honestly one of the best distro's I've ever used. everything just works, rarely any issues.

**Reply by Xanhomey** [reply]
Thanks! Works on Fedora 42.

**Comment by s4ln1x** [comment]
I had to uninstall as zed is not working on rhel9, was expecting for easy steps, but it seems there is no uninstall script yet.

**Comment by CodeWithAlamin** [comment]
How come they say it's a vscode killer while it doesn't even have an efficient uninstaller!?

**Reply by anoduck** [reply]
AND requires a Vulkan compatible GPU driver, and a GitHub Copilot Membership.

**Reply by anoduck** [reply]
@levibassey Don't hate.

**Reply by The-Maize** [reply]
thing is, it has potential, i agree it needed an uninstaller from the get go. 
How ever, im guessing you watched a youtuber that said that as i distinctly remember seeing that ages ago and its not a "vscode killer", that is like saying rust is a c++ killer. its not. its just different. remember that a recommendation is exactly that (especially from a youtube point of view), and their needs probably are completely different to yours. chances are they dont even work day to day on the same language you use.  so always take their recommendations with a grain of salt. this software is fast, slim, customizable (though most "extensions are themes and that really urked me.) and easy to use. it doesnt suit me due to the lack of addons (specifically language support). ill stick with vscode and rider for now. ill check back in on this in a few years when its matured past this point. They really need to separate themes from addons spent over an hour trying to sift through them to find decent actual addons.

**Reply by mandarvaze** [reply]
> AND requires ... GitHub Copilot Membership.

Latest version comes with **free** Claude Sonnet 3.5 (One needs to login, so they can keep track of credits)

**Reply by anoduck** [reply]
@The-Maize I just did a quick search on DuckDuckGo and found numerous blog posts, videos, and sites that employ use of the catchphrase "VScode Killer" in reference to Zed editor. So, by convention of popularity, it appears that this catchphrase has been adopted, at least by a minority, as an unofficial slogan for the editor. It's use and validity is rather mote at this point.

My agreement with criticism of using the aforementioned catchphrase is the lack of an effective uninstallation script (now resolved), requirement of a Vulcan compatible graphics card (still a dealbreaker), and requirement for paid access to copilot (also, now resolved). 

I object to the down voting of my statement, because it was relating a factual observation from personal experience. What was said was not false or invalid, nor was it rude or insensitive. It related facts acquired first hand about software requirements that remained unknown until the editor had already been installed, and the process of troubleshooting why the editor was not running had already begun. These facts would have made a critical difference in the decision to install the editor to begin with, and it was regretful they were not made more obvious beforehand.

Down voting valid and informational comments that relate to a user's experience with software is not only irresponsible, it directly conflicts with the very nature of software development itself, as it is counterproductive to the ability of developers to acquire valid and realistic user feedback.

@mandarvaze The purpose behind Open Source Technology is to reduce the barriers preventing access and acquisition of technology in order to benefit technological knowledge and computational science as a whole. Relinquishing one's private information in order to hurdle a barrier preventing access to technology is an exchange that should not have to be foregone. As, it opens up the user to be vulnerable to breaches in security that would result in the exposure of private to the public. Which is why, IMHO, I personally do not believe Open Source projects should entertain.

**Reply by The-Maize** [reply]
i agree not sure why you were down voted there, what you said was reasonable. To be honest i think the down vote button causes more issues than it does solving them. i think there should be a requirement to explain why you down voted  or up voted. this would prevent people from misinterpreting the meaning of their particular up vote or down vote, it could be they disagreed with you for a particular reason, or just didn't like the way you delivered it. personally i found what you said totally fine, but others interpretation is likely different. this should be considered before deciding they may be delivering some sort of hate. That is the problem with any social platform that involves text instead of face to face discussion, it becomes interpreted by a static representation of what should be interpreted dynamically, based on perceived body language, facial expression and vocal delivery.  With that in mind it could benefit you to look at up or down votes more as an indicator but less as a direct attack on what you said.  

Yeah i cant see it "killing" vs code. for me that sort of terminology makes little sense, there are many IDE's and text editors out there, they have their strengths and weaknesses, personally my favourite is Kate, unfortunately it lacks support in pretty much everything lol, and lsp's can be a bit tedious to get working correctly and sometimes they stop for no reason at all. but its still my favourite, even if i don't use it. VS codes ability is not met in this editor for the tasks that i do, though it was more visually appealing . That doesn't mean it isn't good. it is, it also has functionality that isn't present by default in VSCode. I don't agree with the minority it should be colloquially known as the "VSCode Killer", instead it should be "a reasonable alternative if you use these languages / functions of an editor". I installed this software tested it, gave it a fair go, but ultimately it fell well short of my needs, not to say it falls short of others needs, it possibly wont. but there is definitely room for massive improvement especially in common language support and add on's.

**Reply by mandarvaze** [reply]
@anoduck "VSCode Killer" moniker generates views/clicks - It is a clickbait.
@The-Maize Nobody will watch youtube video with title "a reasonable alternative if you use these languages"
:laughing: 

TBH, I installed zed for Free Claude Sonnet 3.5. But integration was underwhelming. I've not uninstalled Zed yet, but not actively using it either.

**Comment by SomeoneToIgnore** [comment]
Thanks to a great community member, almost 5 months later we've got a bash script that is able to uninstall things.
See https://github.com/zed-industries/zed/pull/21213 for details, further improvements can be landed upon that PR if needed.

To note on the web uninstall script posted above, it is relatively outdated and does not delete everything, in particular, ssh-related files and user configs, ergo bad to use.

**Comment by KTibow** [comment]
TL;DR: `zed --uninstall`

**Reply by Bikash9609** [reply]
Works, thanks!
